contracts/contracts/RequestManager.sol

```diff
@@ -366,34 +375,38 @@ contract RequestManager is Ownable {
         Claim storage claim,
         address depositReceiver
     ) private {
-        collectedBeamerFees += request.beamerFee;
-
         emit DepositWithdrawn(
             claim.requestId,
             depositReceiver
         );
 
+        collectedProtocolFees[request.sourceTokenAddress] += request.protocolFeeToken;
+
         IERC20 token = IERC20(request.sourceTokenAddress);
-        token.safeTransfer(depositReceiver, request.amount);
+        token.safeTransfer(depositReceiver, request.amount + request.lpFeeToken);
         request.depositReceiver = depositReceiver;
 
-        (bool sent,) = depositReceiver.call{value: request.lpFee}("");
+        (bool sent,) = depositReceiver.call{value: request.lpFeeGas}("");
         require(sent, "Failed to send Ether");
     }
 
-    function withdrawbeamerFees() external onlyOwner {
-        require(collectedBeamerFees > 0, "Zero fees available");
-
-        uint256 feeAmount = collectedBeamerFees;
-        collectedBeamerFees = 0;
+    function withdrawProtocolFeeToken(address tokenAddress, address recipient) external onlyOwner {
+        uint256 amount = collectedProtocolFees[tokenAddress];
+        require(amount > 0, "Protocol fee is zero");
+        collectedProtocolFees[tokenAddress] = 0;
 
-        (bool sent,) = msg.sender.call{value: feeAmount}("");
-        require(sent, "Failed to send Ether");
+        IERC20 token = IERC20(tokenAddress);
+        token.safeTransfer(recipient, amount);
```

Comment:
> Just out of curiosity, why is the feature implemented that it can have an arbitrary recipient?

I thought it would be nice to provide that flexibility and thereby avoid a potential extra transaction if the owner decides to forward the fee somewhere else (e.g. a DAO).